examples/expressions/13_timeseries_forecasting.py

```diff
@@ -0,0 +1,488 @@
+
+
+# The "base" variant of the pipeline uses the default |TableVectorizer|, with
+# no hyperparameter search.
+
+
+vectorized = X.skb.apply(TableVectorizer())
+predictions_base = vectorized.skb.apply(default_ridge_pipeline, y=y)
+search_base = predictions_base.skb.get_grid_search(fitted=True, cv=ts)
+
+# %%
+# We can observe the results directly using ``.detailed_results``. Unsurprisingly,
+# the results are not very good. In this case, performing a GridSearch is not
+# necessary, but we will do it for consistency with the next steps.
+
+
+search_base.detailed_results_
+
+
+# %%
+# Prediction with periodic encoders and hyperparameter optimization
+# -----------------------------------------------------------------
+#
+# The |DatetimeEncoder| can generate additional periodic features, which are
+# particularly useful for linear models such as Ridge. Periodic features are off
+# by default, but they can be enabled by setting the
+# ``periodic encoding`` parameter to either  ``circular`` or ``spline``,
+# for trigonometric functions or B-Splines respectively.
+# Periodic features are added for each part of the date: hour in day, day in week,
+# day in month, month in year.
+# For our second pipeline, we define a grid of hyperparameters using the skrub
+# ``choose_*`` functions. In this case, we use |choose_bool| and |choose_from|,
+# as we are not interested in tweaking numerical parameters.
+# The ``choose_*`` functions allow to easily prepare a grid of parameters to use
+# to perform hyperparameter optimization.
+# Here, we use them to turn on and off the ``weekday`` and ``total_seconds`` flag,
+# and to select the specific periodic encoder to use.
+
+
+from skrub import DatetimeEncoder
+
+datetime_encoder = DatetimeEncoder(
+    add_weekday=skrub.choose_bool(name="weekday"),
+    add_total_seconds=skrub.choose_bool(name="total_seconds"),
+    periodic_encoding=skrub.choose_from(
+        [
+            "spline",
+            "circular",
+            None,
+        ],
+        name="periodic_encoding",
+    ),
+)
+
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import KBinsDiscretizer
+
+import skrub.selectors as s
+
+numeric_encoder = make_pipeline(
+    skrub.choose_from(
+        {
+            "kbins": KBinsDiscretizer(n_bins=5, encode="ordinal", strategy="uniform"),
+            "passthrough": "passthrough",
+        },
+        name="numeric_encoder",
+    ),
+    StandardScaler(),
+    SimpleImputer(),
+)
+
+
+vectorized = X.skb.apply(datetime_encoder, cols=s.any_date()).skb.apply(
+    numeric_encoder, cols=s.numeric()
+)
+predictions_enc = vectorized.skb.apply(default_ridge_pipeline, y=y)
+
+
+# Note that, in general, randomized search should be used instead of grid search.
+# In this case, grid search is fine as we are interested in a grid of categorical
+# values.
+
+
+search_enc = predictions_enc.skb.get_grid_search(fitted=True, cv=ts)
```

Comment:
I'm wondering if we could have a way to drop the constant feature created by the `KBinsDiscretizer`, e.g. a `VarianceThreshold`